### DISCUSSION TITLE
How to implement WalletConnect v2 with wagmi v0.12 and Web3Modal?

### DISCUSSION THREAD

**Discussion by MidnightLightning** [WalletConnect/walletconnect-monorepo#2649] [Developer Support] [answered]
Originally posted [over here](https://github.com/WalletConnect/walletconnect-docs/issues/762), and then I realized there's this category specifically for v2 upgrade questions.

 My situation is I use `wagmi` v0.11, and don't want to upgrade to `viem` right now, so plan to stay on `wagmi` v0.12, and using the Web3Modal libraries to do so. However the only guides I can find for Web3Modal are to also upgrade to `wagmi` v1.2 (with a `viem` upgrade).

Is it possible to use `wagmi` v0.12 and some version of Web3Modal to accomplish this upgrade to WalletConnect v2?

**Comment by glitch-txs** [comment]
hey, wagmi has added an ethers adapter if you want to use ethers: https://wagmi.sh/react/ethers-adapters

Otherwise I'd suggest going some commits back on the examples repo, if you have issues with this let me know :)
https://github.com/WalletConnect/web3modal-examples

**Comment by MidnightLightning** [comment]
@glitch-txs 
> Otherwise I'd suggest going some commits back on the examples repo

No, this unfortunately doesn't work. The `wagmi` example was created as part of https://github.com/WalletConnect/web3modal-examples/pull/7, and it was already using `@wagmi/core` version 1.0.1:

https://github.com/WalletConnect/web3modal-examples/pull/7/files#diff-f65c75a48348814bf4029736277ecab67af049f05e3f9e6595cbe076188816bdR11

**Comment by MidnightLightning** [comment]
@glitch-txs 
> wagmi has added an ethers adapter if you want to use ethers: https://wagmi.sh/react/ethers-adapters

Those are "viem → ethers.js adapters"; they are only one-way, and allow taking two specific parts of `viem` ("Public Client" and "Wallet Client") and converting them to the `ethers` equivalents ("Provider" and "Signer"), in case something needs that structure. It does not allow using `ethers` with `wagmi` v1.0. And since the latest versions of Web3Modal are expecting `viem`, it wouldn't be needed if using the latest versions of those libraries together.

**Comment by glitch-txs** [comment, marked as answer]
Hey, just added this PR with the old example from old commit, tested and works fine
https://github.com/WalletConnect/web3modal-examples/pull/20

**Reply by glitch-txs** [reply]
keep in mind you'll have to use old versions of Web3modal (v2). Otherwise you could also use the WalletConnect wagmi's connector which supports v2 too
